Account.java

```diff
@@ -4,14 +4,78 @@
 	
 	private String accountName;
 	private double openingBalance;
+	private double currentBalance;
+	private double amount;
+	private String payee;
+
+
+	//constructor
+	public Account() {
+
+	}
+
 
 	public Account(String accountName, double openingBalance) {
 		this.accountName = accountName;
 		this.openingBalance = openingBalance;
 	}
-	
+
+
+
 	public String toString() {
 		return (accountName + ": " + openingBalance);
 	}
 
+
+	//account = new Account("Main", 100); // account object of the class Account
+
+	// internal transfer out of account, update balance
+	public void internalTransferOut(double Amount, String accountName){
+		this.amount = Amount;
+		this.accountName = accountName;
+		currentBalance = openingBalance - amount;
+	}
+
+	// internal transfer into account, update balance
+	public void internalTransferIn(double Amount, String accountName){
+		this.amount = Amount;
+		this.accountName = accountName;
+		currentBalance = openingBalance + amount;
+	}
+
+
+	public double Transfer(double Amount, String From, String To){
+		this.accountName = From;
+		internalTransferOut(amount, From);
+		this.accountName = To;
+		internalTransferIn(amount, To);
+		return currentBalance;
+
+	}
+
+	// set external payee
+	private void setPayee(String Payee){
+		this.payee = Payee;
+	}
+	// set amount for external payment
+	private void setAmount(Double Amount){
+		this.amount = Amount;
+	}
+
+	// make payment to external payee, deduct money transferred from account balance
+	public void externalTransferOut(String accountName, String Payee, double Amount){
+
+		this.accountName = accountName;
+		setPayee(Payee);
+		setAmount(Amount);
+		currentBalance = openingBalance - amount;
+
+
+	}
```

Comment:
Since the accounts are stored in a Customer class I think it would make more sense to do the transfers there. Accessing accounts through context (this keyword) is pretty complicated and I'm afraid it might not work as we intend to. Consider creating transfer method on a customer, which takes three parameters `from`, `to`, `amount`, the method obviously has to check if there is enough balance to complete the transfer and then perform the transfer of money. 

By looking at the main method I think the methods to pay money and move money was meant to stay in the newbank.java class, so disregard the part above and try to put the method in the main. In the main you should be able to locate currentCustomer object and then perform the transfer between the accounts with the mentioned three parameters. Let me know if it's clear enough otherwise, we can try to make a call and I can explain it better.

Reply:
Hi Lukas,
Thanks for reviewing.
To clarify – are you saying that it would be better to remove the transfer methods from the account class and move them to the NewBank class? In your message above you initially suggest to move them to the customer class, but looking at your second paragraph, am I right in saying just to move them to the NewBank class instead?
I think a call may be a good idea if possible, as I’m not sure I’m clear on what is required. I’m free in the next couple of hours, but after that am not free again till our scheduled meeting tomorrow.
Thanks 
Rohan

Reply:
Hi Rohan, sorry I've missed your reply. I left the first paragraph there so you better understand my train of thoughts. I think moving it to the newbank.java will be better as that is probably where the creator of the code meant to place the code. It will also make it easier to implement I believe. We can take a look later today to the possible approach how to do it. Overall I believe it should be pretty straightforward. reach out if you need a clarification.